lib/cdo/web_purify.rb

```diff
@@ -23,24 +51,33 @@ def self.find_potential_profanities(text, language_codes = ['en'])
       map {|code| ISO_639_1_TO_WEBPURIFY[code] || code}.
       uniq.
       join(',')
-    url = "http://api1.webpurify.com/services/rest/" \
-      "?api_key=#{CDO.webpurify_key}" \
-      "&method=webpurify.live.return" \
-      "&text=#{CGI.escape(text)}" \
-      "&lang=#{language_codes}" \
-      "&format=json"
-    result = JSON.
-      parse(
-        URI.parse(url).
-        open(
-          open_timeout: DCDO.get('webpurify_tcp_connect_timeout', 5),
-          read_timeout: DCDO.get('webpurify_http_read_timeout', 10)
-        ).
-        read
-      )
-
-    expletive = result['rsp'] && result['rsp']['expletive']
-    return nil unless expletive
-    expletive.is_a?(Array) ? expletive : [expletive]
+
+    chunks = split_text(text)
+    expletives = []
+    Net::HTTP.start(API_ENDPOINT.host, API_ENDPOINT.port, CONNECTION_OPTIONS) do |http|
+      chunks.each do |chunk|
+        request = Net::HTTP::Post.new(API_ENDPOINT)
+        form_data = [
+          ['api_key', CDO.webpurify_key],
+          ['format', 'json'],
+          ['lang', language_codes],
+          ['method', 'webpurify.live.return'],
+          ['text', chunk]
+        ]
+        request.set_form form_data, 'multipart/form-data'
```

Comment:
My understanding is if you want to submit form data through `Net::HTTP`, you have to create the request (`Post.new`) and attach the form data (`set_form`) in separate steps. The `Net::HTTP` library has some other shortcut methods like `post_form` (see documentation [here](https://apidock.com/ruby/Net/HTTP/post_form/class)), but it wasn't getting me what I wanted from WebPurify, I think because the default encoding is `application/x-www-form-urlencoded`.